src/mapping/config/MappingConfigurationTypes.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+namespace precice::mapping {
+
+/// How to handle the polynomial?
+/**
+ * ON: Include it in the system matrix
+ * OFF: Omit it altogether
+ * SEPARATE: Compute it separately using least-squares QR.
+ */
+enum class Polynomial {
+  ON,
+  OFF,
+  SEPARATE
+};
+
+enum class Preallocation {
+  OFF,
+  COMPUTE,
+  ESTIMATE,
+  SAVE,
+  TREE
+};
+
+enum class BasisFunctions {
```

Comment:
Why plural? :wink: 
Plural names look like a set of options when passing them to a function.